sdk/internal/recording/recording_test.go

```diff
@@ -517,3 +516,45 @@ type RecordingFileStruct struct {
 type Entry struct {
 	RequestUri string `json:"RequestUri"`
 }
+
+func TestLiveModeOnly(t *testing.T) {
+	LiveOnly(t)
+	if GetRecordMode() == modePlayback {
+		t.Fatalf("Test should not run in playback")
+	}
+}
+
+func TestSleep(t *testing.T) {
+	start := time.Now()
+	Sleep(time.Second * 5)
+	duration := time.Since(start)
+	if GetRecordMode() == modePlayback {
```

Comment:
This might not be an issue with Go's version of this, but my experience with 'timer' related functions is that they're not always as exact as you'd want.

Given that, being too exact with the time limit might make this test flaky.

I think what you really care about here is not the length of time (which honestly you can't screw up) and more that it definitely called sleep (and the time doesn't matter as much). So anything > 1 second, for instance, could be considered 'success'.